### PR TITLE
Imporve Coverage while eliminating a block of fragile boilerplate from the codebase addressing comment.

### LINKED ISSUE #4788: Improve Code Coverage for LibKiwixBook and MetaLinkNetworkEntity.
**Type of Test**
* [x] Unit Test
* [ ] UI Test

**Particular area and files to be tested**
Core entity classes in the `core` module:
- `org.kiwix.kiwixmobile.core.entity.LibkiwixBook`
- `org.kiwix.kiwixmobile.core.entity.MetaLinkNetworkEntity`

**Particular Behaviour to be tested**
- `LibkiwixBook`: Test all properties of getter and setter.
- `MetaLinkNetworkEntity`: verify hashHandles edge cases like unknow hash null similar for pieceHashes

Image

### PR DESCRIPTION
**Pull request by atharvyadav22**
Fixes #4788

Improved Coverage for `LibKiwixBookMark ` and `MetaLinkNetworkEntity` , resolved a very old TODO comment: that simplified `MetaLinkNetworkEntityTest` by reducing boilerplate legacy code.

### CODE REVIEW

**Comment by atharvyadav22** [comment]
@MohitMaliFtechiz 
I was planning to add an edge case for what if XML was corrupted in the sense like no file or urls so the issue is this field is always required 

```
 @field:Element
  var file: FileElement? = null
```

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz
> I was planning to add an edge case for what if XML was corrupted in the sense like no file or urls so the issue is this field is always required

@atharvyadav22 These edge cases should be covered in this :)

**Comment by atharvyadav22** [comment]
> > @MohitMaliFtechiz
> > I was planning to add an edge case for what if XML was corrupted in the sense like no file or urls so the issue is this field is always required
> 
> @atharvyadav22 These edge cases should be covered in this :)

actually for that i need to change that field to `required= false` should i do ? i wanted to ask this as this is very important part and app might crash if such xml exists.

**Comment by MohitMaliFtechiz** [comment]
@atharvyadav22 No `required=false` makes it optional, and it will lead us to future issues. We should not do that. If we can do it without changing the property, then only we should do that. Otherwise we should find some other way to test it.

**Comment by atharvyadav22** [comment]
> @atharvyadav22 No `required=false` makes it optional, and it will lead us to future issues. We should not do that. If we can do it without changing the property, then only we should do that. Otherwise we should find some other way to test it.

yes that's why i asked otherwise everything is upto the code
like code handles that scenario by not allowing that

**Comment by MohitMaliFtechiz** [comment]
> yes that's why i asked otherwise everything is upto the code
like code handles that scenario by not allowing that

No, we should not do that because this is a mandatory element, and it comes from an online source where it is already handled. If it gets corrupted, the XML parser will throw an error, so we don’t need to handle this scenario on our side. We should avoid over-testing cases that don’t make sense or don’t impact user behavior 🙂.

File: core/src/test/java/org/kiwix/kiwixmobile/core/entity/LibKiwixBookTest.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+class LibKiwixBookTest {
+  @AfterEach
+  fun tearDown() {
+    unmockkAll()
+  }
+
+  @Nested
+  inner class PropertyFallbacks {
+    @Test
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/entity/LibKiwixBookTest.kt:56]: @atharvyadav22 From previous to this one, the code coverage is almost similar. That means which scenarios we are testing here. Those are already covered. I understand that the Book object is involved in this, which is native and can not be tested directly. I have made the changes in this test case to cover more scenarios. 

Previous coverage:

Image


New coverage:

Image

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/kiwix/kiwixmobile/core/entity/LibKiwixBookTest.kt (hunk 1 above)
Comment (atharvyadav22) [reply core/src/test/java/org/kiwix/kiwixmobile/core/entity/LibKiwixBookTest.kt:56]: > @atharvyadav22 From previous to this one, the code coverage is almost similar. That means which scenarios we are testing here. Those are already covered. I understand that the Book object is involved in this, which is native and can not be tested directly. I have made the changes in this test case to cover more scenarios.
> 
> Previous coverage:
> 
> Image
> New coverage:
> 
> Image

@MohitMaliFtechiz actually there was an old TODO comment which i solved and yes for that Libkiwix there was nothing much so i didn't do on purpose just to simply increase coverage only which were essential i tackled those.

**Review by MohitMaliFtechiz — APPROVED** [review]

File: core/src/test/java/org/kiwix/kiwixmobile/core/entity/LibKiwixBookTest.kt (hunk 1 above)
Comment (MohitMaliFtechiz) [reply core/src/test/java/org/kiwix/kiwixmobile/core/entity/LibKiwixBookTest.kt:56]: @atharvyadav22 That's good. But at least we should test the LibKiwixBook class on all things to ensure how this behaves when nativeBook is passed. I have made the changes in code, you can have a look :)
